HiP-ThumbnailService/Controllers/ThumbnailsController.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using PaderbornUniversity.SILab.Hip.ThumbnailService.Arguments;
+using PaderbornUniversity.SILab.Hip.ThumbnailService.Utility;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Processing;
+using SixLabors.Primitives;
+
+namespace PaderbornUniversity.SILab.Hip.ThumbnailService.Controllers
+{
+    [Route("api/[controller]")]
+    [Authorize]
+    public class ThumbnailsController : Controller
+    {
+        private readonly DirConfig _dirConfig;
+        private readonly SizeConfig _sizeConfig;
+        private readonly EndpointConfig _endpointConfig;
+
+        /// <summary>
+        /// This dictionary is used for synchronizaing requests for the same id
+        /// </summary>
+        private static readonly ConcurrentDictionary<string, SemaphoreSlim> LockDictionary = new ConcurrentDictionary<string, SemaphoreSlim>();
+
+        public ThumbnailsController(IOptions<DirConfig> uploadConfig, IOptions<SizeConfig> sizeConfig, IOptions<EndpointConfig> endpointConfig)
+        {
+            _dirConfig = uploadConfig.Value;
+            _sizeConfig = sizeConfig.Value;
+            _endpointConfig = endpointConfig.Value;
+        }
+
+        [ProducesResponseType(204)]
+        [HttpDelete]
+        public async Task<IActionResult> Put(string url)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+            
+
+            var encodedId = Convert.ToBase64String(Encoding.UTF8.GetBytes(url));
+            var folderPath = Path.Combine(_dirConfig.Path, encodedId);
+
+            var semaphore = LockDictionary.GetOrAdd(encodedId, new SemaphoreSlim(1));
+
+            await semaphore.WaitAsync();
+            try
+            {
+                //clear folder
+                if (Directory.Exists(folderPath))
+                {
+                    var folder = new DirectoryInfo(folderPath);
```

Comment:
Why not just `Directory.Delete(folderPath)`?

Reply:
Directory.Delete only deletes empty Directories, but DirectoryInfo.Deletes everything recursively

Reply:
To prevent an DirectoryNotFoundException

Reply:
I mean, why `new DirectoryInfo(folderPath).Delete(true)` instead of `Directory.Delete(folderPath, true)`?